pkg/auth/routes.go

```diff
@@ -43,14 +43,18 @@ func NewRouter(p *deps.RootProvider, configSource *configsource.ConfigSource) ht
 
 	router.Health(p.RootHandler(newHealthzHandler))
 
-	rootChain := httproute.Chain(
+	noProjectChain := httproute.Chain(
 		p.RootMiddleware(newOtelMiddleware),
 		p.RootMiddleware(newPanicMiddleware),
 		p.RootMiddleware(newBodyLimitMiddleware),
 		p.RootMiddleware(newSentryMiddleware),
 		httproute.MiddlewareFunc(httputil.XContentTypeOptionsNosniff),
 		httproute.MiddlewareFunc(httputil.PermissionsPolicyHeader),
 		httproute.MiddlewareFunc(httputil.XRobotsTag),
+	)
+
+	rootChain := httproute.Chain(
+		noProjectChain,
 		MakeWebAppRequestMiddleware(p, configSource, newWebAppRequestMiddleware),
 		p.Middleware(newContextHolderMiddleware),
```

Comment:
I think ContextHolderMiddleware is needed in noProjectChain too.

Reply:
Currently the preview route does not use anything inside it.

Reply:
I found that the current no project routes are missing CSP, I created another issue to configure CSP of the no project route:
https://linear.app/authgear/issue/DEV-2705/configure-csp-for-preview-route

Reply:
The CSP thing should require CSP nonce which depends on ContextHolderMiddleware.